packages/dds/tree/src/feature-libraries/schema-index/codec.ts

```diff
@@ -57,9 +94,32 @@ function decode(f: Format): TreeStoredSchema {
 /**
  * Creates a codec which performs synchronous monolithic encoding of schema content.
  */
-export function makeSchemaCodec(options: ICodecOptions): IJsonCodec<TreeStoredSchema, Format> {
-	return makeVersionedValidatedCodec(options, new Set([schemaFormat.version]), Format, {
-		encode: (data: TreeStoredSchema) => encodeRepo(data),
-		decode: (data: Format) => decode(data),
-	});
+export function makeSchemaCodec(
```

Comment:
I don't think you want to use makeVersionedValidatedCodec. When encoding schema in v2, you don't want a codec which will error when trying to read v1 data (which is what this will give you)

I think you want to use makeVersionDispatchingCodec. I haven't used it, but it looks like the thing you want for making a coded which supports multiple formats.